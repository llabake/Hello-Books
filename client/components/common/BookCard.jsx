import React from 'react';
import { Link } from 'react-router-dom';
import { bookDefaultImage } from '../../helpers/utils';

import leanstart from '../../media/lean start.jpg';

const BookCard = (props) => {
  const { book } = props;
  return (
    <div className="book col s12 m3 l3">
      <div className="card" style={{ minHeight: '250px', maxHeight: '250px' }}>
        <div className="card-image waves-effect waves-block waves-light" style={{ maxHeight: '175px', minHeight: '170px', paddingTop: '10px' }}>
          <Link to={'/book/' + book.id}>
            <img src={book.image || bookDefaultImage}
              style={{ maxHeight: '170px', minHeight: '165px' }}
              alt="Book Image"
              title={book.title} 
            />
          </Link>
        </div>
        <div className="card-content"
          style={{ position: 'absolute', bottom: 0, width: '100%' }}>
          <p><Link to={'/book/' + book.id} className="btn">Details</Link></p>
        </div>
      </div>
      <div className="book-info center-align">
        <div className="book-title"><span>{book.title} </span></div>
        <div className="book-author">by <span>{book.author}</span></div>
      </div>
    </div>

  );
}

export default BookCard;