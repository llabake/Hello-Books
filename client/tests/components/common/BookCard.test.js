import React from 'react';

import BookCard from '../../../components/common/BookCard';
import { book2 } from '../../mocks/bookData';

test('The header component mounts correctly if all props it needs are passed into it.', () => {
  const props = {
    book: book2,
  }
  const wrapper = shallow(<BookCard {...props} />)
  // console.log(wrapper.debug(), 'uhm')
  expect(wrapper.exists()).toBe(true);

  // expect(wrapper.find('img').length).toBe(1);
  expect(wrapper).toMatchSnapshot();  
})