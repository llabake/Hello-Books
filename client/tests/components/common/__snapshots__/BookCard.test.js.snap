// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The header component mounts correctly if all props it needs are passed into it. 1`] = `
<div
  className="book col s12 m3 l3"
>
  <div
    className="card"
    id="book-undefined"
    style={
      Object {
        "maxHeight": "250px",
        "minHeight": "250px",
      }
    }
  >
    <div
      className="card-image waves-effect waves-block waves-light"
      style={
        Object {
          "maxHeight": "175px",
          "minHeight": "170px",
          "paddingTop": "10px",
        }
      }
    >
      <Link
        replace={false}
        to="/book/5"
      >
        <img
          alt="Book Image"
          src="https://res.cloudinary.com/sardaunan/image/upload/v1518354032/book-default-image_e8nj1p.png"
          style={
            Object {
              "maxHeight": "170px",
              "minHeight": "165px",
            }
          }
          title="Everyday is for the thief"
        />
      </Link>
    </div>
    <div
      className="card-content"
      style={
        Object {
          "bottom": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <p>
        <Link
          className="btn test-btn"
          replace={false}
          to="/book/5"
        >
          Details
        </Link>
      </p>
    </div>
  </div>
  <div
    className="book-info center-align"
  >
    <div
      className="book-title"
    >
      <span>
        Everyday is for the thief
         
      </span>
    </div>
    <div
      className="book-author"
    >
      by 
      <span>
        Teju Cole
      </span>
    </div>
  </div>
</div>
`;
